src/gamescope/wayland/metrics/mango_app_reader.rs

```diff
@@ -0,0 +1,44 @@
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+
+use gamescope_wayland_client::mangoapp::{message::MangoAppMsgV1, queue::MangoAppMsgQueue};
+
+pub struct MangoAppMsgQueueReader {
+    queue: MangoAppMsgQueue,
+    rx: std::sync::mpsc::Receiver<()>,
+    tx: tokio::sync::watch::Sender<Option<MangoAppMsgV1>>,
+}
+
+impl MangoAppMsgQueueReader {
+    pub fn new(
+        rx: std::sync::mpsc::Receiver<()>,
+        tx: tokio::sync::watch::Sender<Option<MangoAppMsgV1>>,
```

Comment:
We should use the unqualified path to make things a bit less verbose.

```suggestion
        tx: watch::Sender<Option<MangoAppMsgV1>>,
```